experimenter/experimenter/nimbus_ui/static/js/edit_branches.js

```diff
@@ -1,58 +1,27 @@
-import {
-  drawSelection,
-  EditorView,
-  highlightActiveLine,
-  highlightSpecialChars,
-  keymap,
-  lineNumbers,
-} from "@codemirror/view";
+import { basicSetup } from "codemirror";
+import { EditorView } from "@codemirror/view";
 import { json, jsonParseLinter } from "@codemirror/lang-json";
 import { linter } from "@codemirror/lint";
-import { autocompletion, closeBrackets } from "@codemirror/autocomplete";
-import {
-  HighlightStyle,
-  syntaxHighlighting,
-  defaultHighlightStyle,
-  foldGutter,
-  bracketMatching,
-  indentOnInput,
-} from "@codemirror/language";
-import {
-  highlightSelectionMatches,
-  searchKeymap,
-  search,
-} from "@codemirror/search";
-import { defaultKeymap, historyKeymap, history } from "@codemirror/commands";
-import { tags } from "@lezer/highlight";
+import { autocompletion } from "@codemirror/autocomplete";
 import { schemaAutocomplete, schemaLinter, fmlLinter } from "./validator.js";
+import {
+  themeCompartment,
+  getThemeExtensions,
+  updateViewTheme,
+  observeThemeChanges,
+} from "./theme_utils.js";
 import $ from "jquery";
 
+const codemirrorViews = [];
```

Comment:
We should use a [WeakSet](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WeakSet) here because otherwise this can keep references to DOM elements that get removed from the page.